inginious/frontend/pages/course_admin/settings.py

```diff
@@ -131,6 +134,28 @@ def define_tags(self, course, data, course_content):
         course_content["tags"] = tags
         self.course_factory.update_course_descriptor_content(course.get_id(), course_content)
 
+    def define_additionnal_fields(self, course, data, course_content):
+        """Additional field definition method"""
+        fields = self.prepare_datas(data, "field")
+        if not isinstance(fields, dict):
+            # prepare_datas returned an error
+            return fields
+
+        # Repair fields
+        for key, field in fields.items():
+            field["type"] = int(field["type"])
+
+            if (field["id"] == "" and field["type"] != 2):
+                return _("Some fields were missing.")
```

Comment:
This is dead code: the ``prepare_datas`` return line filters out any field that has no id, as per documentation, this is the way to remove them from the list (no delete button)